builtin/providers/openstack/resource_openstack_image_v2.go

```diff
@@ -0,0 +1,494 @@
+package openstack
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imagedata"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func resourceImageV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceImageV2Create,
+		Read:   resourceImageV2Read,
+		Update: resourceImageV2Update,
+		Delete: resourceImageV2Delete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
```

Comment:
This should be removed since `id` shouldn't be a visible attribute. `d.SetId(image.ID)` effectively sets the resource's "internal" id.